.evergreen/config.yml

```diff
@@ -2027,102 +2027,102 @@ axes:
   - id: os-ssl-32
     display_name: OS
     values:
-      - id: "windows-64-go-1-20"
+      - id: "windows-64"
```

Comment:
I removed the Go version from the OS matrix ID because it causes us to lose Evergreen build history every time we upgrade Go. That pattern was started in [this PR](https://review.gerrithub.io/c/mongodb/mongo-go-driver/+/425510), which doesn't provide any obvious insight into why it was changed. Unless it causes some issue, we should not change the OS matrix ID when we upgrade Go.